### PR TITLE
Add an option to build LLVM with CMake/Ninja

### PR DESCRIPTION
**Pull request by Keno**
Makes for faster incremental builds when developing on LLVM.
This version already addresses some of @tkelman's comments who likes to review changes before they're ever pull requests :racehorse: .

### CODE REVIEW

File: deps/Makefile [hunk 1]
```diff
@@ -51,6 +51,17 @@ ifneq ($(strip $(CMAKE_CXX_ARG)),)
 CMAKE_COMMON += -DCMAKE_CXX_COMPILER_ARG1="$(CMAKE_CXX_ARG)"
 endif
 
+# For now this is LLVM specific, but I expect it won't be in the future
+ifneq ($(LLVM_USE_CMAKE),)
+ifeq ($(CMAKE_GENERATOR),Ninja)
+CMAKE_GENERATOR_COMMAND = -G Ninja
+else ifeq ($(CMAKE_GENERATOR),make)
+CMAKE_GENERATOR_COMMAND = -G "Unix Makefiles"
```
Comment (tkelman) [inline deps/Makefile]: may need to be "MSYS Makefiles" for the windows build, but can test that later

File: src/Makefile [hunk 2]
```diff
@@ -27,7 +27,11 @@ endif
 # In LLVM >= 3.4, --ldflags has only options, and --system-libs has the libraries.
 LLVMLINK = $(call exec,$(LLVM_CONFIG) --ldflags) $(call exec,$(LLVM_CONFIG) --libs) $(call exec,$(LLVM_CONFIG) --ldflags) $(call exec,$(LLVM_CONFIG) --system-libs 2> /dev/null)
 ifeq ($(USE_LLVM_SHLIB),1)
+ifeq ($(LLVM_USE_NINJA),1)
```
Comment (tkelman) [inline src/Makefile]: `LLVM_USE_CMAKE` ?

File: ui/Makefile [hunk 3]
```diff
@@ -21,7 +21,11 @@ DOBJS = $(SRCS:%=%.dbg.obj)
 DEBUGFLAGS += $(FLAGS)
 SHIPFLAGS += $(FLAGS)
 ifeq ($(USE_LLVM_SHLIB),1)
+ifeq ($(LLVM_USE_NINJA),1)
```
Comment (tkelman) [inline ui/Makefile]: `LLVM_USE_CMAKE` ?

**Comment by tkelman** [comment]
sorry, saw the new branch on a fetch and couldn't resist - we will need to prepare for this anyway since upstream is planning on deprecating autotools before too much longer

File: deps/Makefile (hunk 1 above)
Comment (Keno) [inline deps/Makefile]: yes, whoever is the first to try this on Windows gets to make that change ;).

**Pull request merged**